src/Process/3dTilesProcessing.js

```diff
@@ -236,20 +236,27 @@ export function pre3dTilesUpdate(context, layer) {
     const HYFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) * hypotenuse / context.camera.width);
     context.camera.preSSE = hypotenuse * (2.0 * Math.tan(HYFOV * 0.5));
 
-    // once in a while, garbage collect
-    if (Math.random() > 0.98) {
+    // Elements removed are added in the layer._cleanableTiles list.
+    // Since we simply push in this array, the first item is always
+    // the oldest one.
+    const now = Date.now();
+    if (layer._cleanableTiles.length
```

Comment:
In `3dTilesProvider`, maybe `_cleanableTiles` could be declared before the `preUpdate` method and this test could be avoided ?

Reply:
This test is to check if the `_cleanableTiles` is empty - if it's empty we skip the garbage collection mechanism.
So I don't really understand your suggestion?

Reply:
Yep I was mistaken, please dismiss my comment.